app/services/applier/form_filler.rb

```diff
@@ -1,69 +1,131 @@
 # frozen_string_literal: true
 
-require 'json'
-# require 'open-uri'
-# require 'htmltoword'
-
-# TODO: Handle job posting becoming closed (redirect or notification on page)
 module Applier
-  # Core class for filling out forms using Capybara
   class FormFiller < ApplicationTask
     include Capybara::DSL
-
-    def initialize(job_application, payload)
-      @job_application = job_application
-      @url = @job_application.job.posting_url
-      @ats = @job_application.job.applicant_tracking_system
-      @fields = JSON.parse(payload)
-      # @user = @job_application.application_process.user
-      @errors = nil
-
-      # include the relevant ATS form_filler module
-      include_ats_module
+    def initialize(payload)
+      @application_form = payload[:form_locator]
+      @fields = payload[:fields]
+      @session = Capybara::Session.new(:selenium)
+      @url = payload[:apply_url]
+      @user_fullname = payload[:user_fullname]
     end
 
     def call
-      return unless processable
+      return false unless processable
 
-      process
-    rescue StandardError => e
-      Rails.logger.error "Error running FormFiller: #{e.message}"
-      nil
+      using_session(@session) do
+        process
+      end
     end
 
     private
 
     def processable
-      @url && @fields && @job_application && @ats
+      @url && @fields && @session
     end
 
     def process
-      p "Hello from FormFiller!"
-      submit_application
+      visit_url
+      fill_application_form
+      click_submit_button
+      verify_submission
+    ensure
+      @session.quit
     end
 
-    def include_ats_module
-      ats_name = @ats.name.gsub(/\W/, '').capitalize
-      module_name = "Ats::#{ats_name}::SubmitApplication"
-      extend Object.const_get(module_name) if Object.const_defined?(module_name)
+    def apply_button
+      find(:css, 'button, a', text: /apply/i, match: :first)
     end
 
-    def submit_application
-      return super if defined?(super)
+    def attach_file_to_application
+      find(@locator).attach_file(@filepath)
+    end
 
-      puts "Write a submit method for #{@ats.name}!"
-      return
+    def click_apply_button
+      apply_button.click
     end
 
-    # def take_screenshot_and_store(session, job_application)
-    #   screenshot_path = Rails.root.join('tmp', "screenshot-#{job_application.id}.png")
-    #   session.save_screenshot(screenshot_path)
+    def click_submit_button
+      sleep 2 # temporary -- just for testing
+      submit_button.click
+    end
 
-    #   file = File.open(screenshot_path)
-    #   job_app = job_application
-    #   job_app.screenshot.attach(io: file, filename: "screenshot-#{job_application.id}.png", content_type: 'image/png')
+    def doc_tmp_file
```

Comment:
This is one to go into more detail on later with JB on how we want to handle files - he was discussing markdown for cover letters for example